src/axom/inlet/inlet_utils.hpp

```diff
@@ -174,6 +182,32 @@ inline bool isCollectionGroup(const std::string& name)
 */
 void markAsStructCollection(axom::sidre::Group& target);
 
+/*!
+*****************************************************************************
+* \brief Adds a ReaderResult to a sidre::Group corresponding to an inlet
+* object, if the retrieval is not successful
+*
+* \param [inout] target The group to tag
+* \param [in] result The retrieval result
+*****************************************************************************
+*/
+void markRetrievalStatus(axom::sidre::Group& target, const ReaderResult result);
+
+/*!
+*****************************************************************************
+* \brief Returns the corresponding retrieval result for a collection depending
+* on whether the collection is empty and whether the corresponding collection
+* in the input file was homogeneous
+*
+* \param [in] none_of_other_type Whether any collection elements were of type
+* other than the requested type
+* \param [in] none_of_requested_type Whether the collection of requested type was empty,
+* i.e., if no elements of the requested type were present
+*****************************************************************************
+*/
+ReaderResult collectionRetrievalResult(const bool none_of_other_type,

```

Comment:
Booleans with negative names can be harder to work with. How about `all_of_other_type` and `all_of_requested_type`? Of course, you'd have to change the logic and call sites. Not sure it's worth it.

Reply:
This is partial my fault... i was confused by the previous names.  They made sense at the call sites but not at the function level... at least to me.